### PR TITLE
100 lighting

### PR DESCRIPTION
**Pull request by Minywire**
## What problem does this pull request address?

One of our core requirements is that we need terrain lighting, we do not currently have any lighting implemented at all.

## How does this pull request solve the problem?

This pull request implements **model** lighting for our engine using the Phong approach of diffuse, ambient, and specular components.

### Things to do after this PR is merged
- Implement the Phong lighting for the terrain
- Refactor the lighting shaders and the OpenGLModel class so that we can load this data in via material files
- Refactor this so that it works with scripting

## What testing has been performed?

- [x] Builds successfully
- [x] Runs successfully

https://github.com/Minywire/Bottlebrush_Engine/assets/51070144/a11e960a-5398-4442-ad11-088dd076312e

Diffuse

https://github.com/Minywire/Bottlebrush_Engine/assets/51070144/34275456-0de7-40f9-ba06-be4f3721314e

### CODE REVIEW

File: Bottlebrush_Engine/Bottlebrush_Engine/RenderEngine/Platform/OpenGL/OpenGLMesh.cpp [hunk 1]
```diff
@@ -51,7 +50,7 @@ void OpenGLMesh::CreateMesh(
     m_Texture->CreateTexture(textureFilePath);
 
     // bind the texture
-    SetTexture(); // @TODO probably not needed? as it needs to bind before draw call
+    SetTexture();
```
Comment (Idraliski) [inline Bottlebrush_Engine/Bottlebrush_Engine/RenderEngine/Platform/OpenGL/OpenGLMesh.cpp:53]: Can we remove this? it isn't needed as `SetTexture()` has to be called before each draw anyways

**Review by Idraliski — CHANGES_REQUESTED** [review]
Other than than 1 I commented on, everything looks good to me. Works on my PC

File: Bottlebrush_Engine/Bottlebrush_Engine/RenderEngine/Platform/OpenGL/OpenGLMesh.cpp (hunk 1 above)
Comment (Minywire) [reply Bottlebrush_Engine/Bottlebrush_Engine/RenderEngine/Platform/OpenGL/OpenGLMesh.cpp]: all done i've removed this

**Review by Idraliski — APPROVED** [review]
Looks good to me

**Pull request merged**